lib/jit_preloader/preloader.rb

```diff
@@ -3,25 +3,44 @@ class Preloader < ActiveRecord::Associations::Preloader
 
     attr_accessor :records
 
-    def self.attach(records)
-      new(records: records.dup, associations: nil).tap do |loader|
-        records.each do |record|
-          record.jit_preloader = loader
+    if Gem::Version.new(ActiveRecord::VERSION::STRING) >= Gem::Version.new("7.0.0")
+      def self.attach(records)
+        new(records: records.dup, associations: nil).tap do |loader|
```

Comment:
Discussed with Wendy, on ActiveRecord Version 7, we can pass in the associations variable for the object, and that call will fail if done on ActiveRecord version 6 or below, it does not accept the associations parameter necessitating the `else` statement.

Reply:
Yes, ActiveRecord below 7.xx does not accept any arguments in initialization method.
[6-0-stable](https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/associations/preloader.rb#L97-L99)
[7-0-stable](https://github.com/rails/rails/blob/7-0-stable/activerecord/lib/active_record/associations/preloader.rb#L96)